### PR TITLE
Extra warning about rolling upgrades and sharding added

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Are `java/cluster-sharding.md` and `scala/cluster-sharding.md` very different? If not might make sense to merge them (and replace `java/cluster-sharding.md` with a symlink) while we're at it.

**Comment by johanandren** [comment]
Very much alike, but it's a long docs-chapter, I'd prefer to not do that right now.

File: akka-docs/src/main/paradox/java/cluster-sharding.md [hunk 1]
```diff
@@ -421,3 +421,13 @@ of entities that are alive in each shard.
 
 The purpose of these messages is testing and monitoring, they are not provided to give access to
 directly sending messages to the individual entities.
+
+## Rolling upgrades
+
+When doing rolling upgrades special care must be taken to not change any of the following aspects of sharding:
+
+ * the `extractShardId` function so that it extracts a different shard id than before for a given message
```
Comment (patriknw) [inline akka-docs/src/main/paradox/java/cluster-sharding.md]: this sentence is unclear, I think. should it return a different shard id or should it not? ;-)

Comment (johanandren) [inline akka-docs/src/main/paradox/java/cluster-sharding.md]: Yeah, I see. I wanted to make a difference between adding new types of messages to it and changing what id it returns from an existing message. I'll see if I can rephrase.

**Review by patriknw — APPROVED** [review]
LGTM, simple and clear

**Pull request merged**